SO- 4890: support includeNull and pretty query parameters

cmark opened the pull request:
Both query parameters work with empty or `true` values. Any other value considered `false` and will not enable the feature.
* `includeNull` - includes all properties of the response model to be
serialized to the JSON representation even if the value is `null`. The
default behavior is to omit `null` values.
* `pretty` - pretty print the output to make it human-readable

Fixes #486.

cmark committed:
feat: support `includeNull` and `pretty` query parameters...

...to configure returned JSON representation

Both query parameter work with empty or `true` values. Any other value
considered `false` and will not enable the feature.
* `includeNull` - includes all properties of the response model to be
serialized to the JSON representation even if the value is `null`. The
default behavior is to omit `null` values.
* `pretty` - pretty print the output to make it human-readable

cmark committed:
fix: objectMapper creation in FhirTest class

apeteri commented on core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/SnowOwlApiConfig.java:
```diff
@@ -120,6 +127,53 @@ public ObjectMapper objectMapper() {
+	}
+
+	private boolean extractBooleanQueryParameterValue(HttpServletRequest request, String queryParameterKey) {
+		String[] values = request.getParameterMap().containsKey(queryParameterKey) ? request.getParameterMap().getOrDefault(queryParameterKey, null) : null;
```
`getOrDefault` already checks for the existence of the key:

```suggestion
		String[] values = request.getParameterMap().getOrDefault(queryParameterKey, null);
```

cmark committed:
refactor: use LoadingCache and BitSet to defer construction of...

...object mappers until not requested and to allow future extensions of
serialization feature configurations.
